com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ModuleHooksAggregator.java

```diff
@@ -82,7 +82,26 @@ private boolean shouldExecuteHook(Map<String, List<String>> alreadyExecutedHooks
     private boolean hasAllPhasesExecuted(Map<String, List<String>> alreadyExecutedHooks, Hook hookToBeExecuted,
                                          List<HookPhase> hookPhasesForCurrentStepPhase) {
         List<HookPhase> executedHookPhasesForHook = getExecutedHookPhasesForHook(alreadyExecutedHooks, hookToBeExecuted.getName());
-        return executedHookPhasesForHook.containsAll(hookPhasesForCurrentStepPhase);
+        return executedHookPhasesForHook.containsAll(getHookPhasesForCurrentStepPhase(hookToBeExecuted, hookPhasesForCurrentStepPhase));
+    }
+
+    @Deprecated
+    private List<HookPhase> getHookPhasesForCurrentStepPhase(Hook hookToBeExecuted, List<HookPhase> hookPhasesForCurrentStepPhase) {
+        boolean isThereOldPhase = hookToBeExecuted.getPhases()
+                                                  .stream()
+                                                  .anyMatch(hookPhase -> HookPhase.getOldPhases()
+                                                                                  .contains(HookPhase.fromString(hookPhase)));
+        if (isThereOldPhase) {
+            return hookPhasesForCurrentStepPhase.stream()
+                                                .filter(hookPhase -> HookPhase.getOldPhases()
```

Comment:
Extract `HookPhase.getOldPhases()
                                                                               .contains(hookPhase)` into method and re-use it below